extension/parquet/parquet_reader.cpp

```diff
@@ -242,7 +244,11 @@ LogicalType ParquetReader::DeriveLogicalType(const SchemaElement &s_ele, bool bi
 		case ConvertedType::INTERVAL:
 			return LogicalType::INTERVAL;
 		case ConvertedType::JSON:
-			return LogicalType::VARCHAR;
+			if (context_ptr) {
+				Catalog::GetEntry<TypeCatalogEntry>(*context_ptr, INVALID_CATALOG, INVALID_SCHEMA, "JSON",
```

Comment:
We're now adding a client context to the `ParquetReader` only for this check - what happens if we don't do this? Since we bake the JSON extension into many clients this is a rather uncommon scenario.

We had previously removed the ClientContext from the ParquetReader on purpose, although the reason for that is now no longer relevant, so I'm not super opposed to this, but it doesn't look particularly elegant from a code perspective imo.

Reply:
If we just return `LogicalType::JSON()` without the `ClientContext`, users will get back the `JSON` type without it actually being available in the catalog.

I'm not opposed to this, but I think it's nicer if users get an error telling them to install the JSON extension. Maybe the binder should check if the types returned by table functions are available in the catalog and throw an error if they are not. What do you think?

Reply:
This situation is already possible by loading a DuckDB file that contains a `JSON` column, no? What happens in that case?

Reply:
That works, you will get back the `JSON` type without it being loaded in the catalog

Reply:
Isn't that the same situation then?